src/index.ts

```diff
@@ -119,17 +119,67 @@ function makePassesFilter(opts: FilterOpts): (msg: Msg) => boolean {
 class threads {
   private readonly ssb: Record<string, any>;
   private readonly isBlocking: (obj: any, cb: CB<boolean>) => void;
+  private readonly isFollowing: ((obj: any, cb: CB<boolean>) => void) | null;
 
   constructor(ssb: Record<string, any>, _config: any) {
     this.ssb = ssb;
     this.isBlocking = ssb.friends?.isBlocking
       ? ssb.friends.isBlocking
       : IS_BLOCKING_NEVER;
+    this.isFollowing = ssb.friends?.isFollowing;
     this.ssb.db.registerIndex(HashtagsPlugin);
   }
 
   //#region PRIVATE
 
+  private maybeOnlyKeepFollowingMessages =
+    (enabled: boolean) => (source: any) =>
+      pull(
+        source,
+        pull.asyncMap((msg: Msg, cb: CB<Msg | null>) => {
+          if (!enabled) {
+            cb(null, msg);
+            return;
+          }
+
+          if (!this.isFollowing) {
+            cb(null, msg);
+            return;
+          }
+
+          function isFollowingCb(err: any, following: boolean) {
+            if (err) cb(err);
+            else if (!following) cb(null, null);
+            else cb(null, msg);
+          }
+
+          const rootMsgKey = getRootMsgId(msg);
+
+          if (rootMsgKey === msg.key) {
+            this.isFollowing(
```

Comment:
We have to remember that the current user is also supposed to have their root posts show on the feed, so that means that early in this function we can check whether `msg.value.author === this.ssb.id` and allow that msg to pass, because I believe `isFollowing` would return false.

Reply:
Does this include contact messages? After some refactoring, noticing that my `threads.public` test fails with the operations (from Lucy's db):

1. Lucy follows Mary
2. Mary creates root post
3. Alice creates root post

The test is expecting just Mary's post to be returned by the method, but it's also returning a thread with the contact message in 1. Is that the expected behavior?

Reply:
Made the assumption that ^ is the expectation, so updated the test to account for contact messages

Reply:
That was the correct assumption, because publicUpdates in the test is considering all the message types. Typically we pass a "filter" to the publicUpdates where we tell it we are only interested in (say) `post` messages, such that the `contact` message would be left out.